src/SignalR/clients/csharp/Client.Core/src/HubConnection.cs

```diff
@@ -1083,6 +1085,25 @@ private async Task InvokeStreamCore(ConnectionState connectionState, string meth
         }
     }
 
+    private static void InjectHeaders(HubInvocationMessage invocationMessage)
+    {
+        // TODO: Change when SignalR client has an activity.
+        // This sends info about the current activity, regardless of the activity source, to the SignalR server.
+        // When SignalR client supports client activities this logic should be updated to only send headers
+        // if the SignalR client activity is created. The goal is to match the behavior of distributed tracing in HttpClient.
+        if (Activity.Current is { } currentActivity)
```

Comment:
This means every single hub method call for the lifetime of the connection will fall under a single span again. We were avoiding that before but this just reintroduces that with more ceremony. Shouldn't we be making a span per call so that you can see how long it took between client starting the request and server getting the request (plus full E2E time etc.)?

Reply:
Everything isn't falling under one span again. `Activity.Current` could change depending upon the context that someone invokes the server.

Creating a client span is useful (it would sit under whatever `Activity.Current` is at that point in time, but I was focusing on propagation in this PR.

Do you feel strongly that a client span should happen in .NET 9?

Reply:
I'm looking at adding a client span now for .NET. I'm going to make it a new PR. It's a fairly large amount of work, and this PR is big enough already.